frontend/javascript/README.md

```diff
@@ -0,0 +1,713 @@
+```js
+undefined
+null
+```
+
+[^other-primitives]: Okay, technically there's also [`bigint`](https://developer.mozilla.org/en-US/docs/Glossary/BigInt) and [`symbol`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol), but most readers probably don't need to worry about those.
+
+### Variables
+You may from time to time want to store values for later use. You can do this using variables.
+```js
+// declares a new variable `variable0` with initial value `undefined`
+let variable0;
+// variables can be reassigned
+variable0 = "hi";
+// js is dynamically typed, so variables can also be reassigned to values with
+// different types
+variable0 = 0;
+// const variables must be initialized at declaration and cannot be reassigned
+// after declaration
+// const variable1; <- not allowed!
+const variable1 = 1;
+// variable1 = 2; <- not allowed!
+
+// variables are scoped to the block they're in. this means that they're only
+// accessible inside the curly braces ({}) they were declared.
+let outerScope = "outer!";
+// outerScope now accessible
+{
+    // outerScope still accessible
+    let innerScope = "inner!";
+    // innerScope now accessible
+}
+// innerScope no longer accessible
+// outerScope still accessible
+```
+
+Sometimes you'll want to do an operation on a variable's value and then store the result back into the variable. JavaScript has a shorthand for this:
+```js
+let foo = 100;
+// instead of writing something like this
+foo = foo + 1;
+// you can instead write this
+foo += 1;
+```
+You use this operation-then-assign syntax with any of the operations mentioned in the previous section.
+
+### Comparisons
+Comparisons let you check whether data structures satisfy some sort of condition.
+
+```js
+// double-equals (==) compares values. this yields true:
+0 == 0
+// but beware! the interpreter will try and convert types to allow for
+// comparison. this can lead to unexpected behavior like this (comparing string
+// and number), which yields true
+"0" == 0
+
+// if you want to do an equals comparison while respecting type differences,
+// you'd use a triple-equals (===) comparison.
+0 === 0 // true
+"0" === 0 // false
+
+// a similar distinction exists for the not-equals comparisons. the following
+// are equivalent, and yield false
+!(0 == 0)
+0 != "0"
+// the following are equivalent, and yield true
+!(0 === "0")
+0 !== "0"
+
+// finally, we have the ordering comparisons. these work as you would expect in
+// any other language, with the caveat that types are always converted (like
+// with ==)
+0 > 0 // false
+0 >= "0" // true
+0 < "0" // false
+0 <= 0 // true
```

Comment:
I didn't talk about the string ordering comparisons since they're not terribly common and I didn't want to type out an explanation of lex order. Should they be touched on?

Reply:
I wouldn't want to explain lex order either! If for some reason lex order comparisons are needed by a member, maybe chuck in a link/note directing readers to an in-depth explanation of it, since it has such a niche use-case?

Reply:
I think I'm comfortable letting people discover it themselves if they every need it. The JS resources listed at the bottom (or Google) should be enough for curious readers.

If anyone ever asks about it in the context of the club, we can add it pretty easily.